python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -930,13 +930,52 @@ def _fast_slow_function_call(
                             "Pandas debugging mode failed. "
                             f"The exception was {e}."
                         )
-    except Exception:
+    except Exception as err:
         with nvtx.annotate(
             "EXECUTE_SLOW",
             color=_CUDF_PANDAS_NVTX_COLORS["EXECUTE_SLOW"],
             domain="cudf_pandas",
         ):
             slow_args, slow_kwargs = _slow_arg(args), _slow_arg(kwargs)
+            if _env_get_bool("CUDF_PANDAS_SLOW_LOG", False):
+                from ._logger import StructuredMessage, logger
+
+                def reprify(arg) -> str:
+                    try:
+                        return repr(arg)
+                    except Exception:
+                        return "<REPR FAILED>"
+
+                module = getattr(slow_args[0], "__module__", "")
+                obj_name = getattr(
+                    slow_args[0], "__name__", type(slow_args[0]).__name__
+                )
+                slow_object = f"{module}.{obj_name}"
+                # TODO: maybe use inspect.signature to map called args and kwargs
+                # to their keyword names
+                called_args = ",".join((reprify(val) for val in slow_args[1]))
+                if len(slow_args) == 3:
+                    fmt_kwargs = ",".join(
+                        f"{kwarg}={reprify(value)}"
+                        for kwarg, value in slow_args[2].items()
+                    )
+                    called_args = ",".join((called_args, fmt_kwargs))
+                    passed_kwargs = {
+                        kwarg: reprify(value)
+                        for kwarg, value in slow_args[2].items()
+                    }
+                else:
+                    passed_kwargs = {}
```

Comment:
Why are we never looking `slow_kwargs`?

Reply:
I think `slow_kwargs` is always empty. Because `kwargs` is getting subsumed by `args` in the `_fast_slow_function_call` calls. I opened #16266  to fix.

Reply:
> I think slow_kwargs is always empty.

Correct. Seems like all calls to `_fast_slow_function_call` internally are of the form `_fast_slow_function_call(something, object, args, kwargs)` so `kwargs` are actually in `slow_args` instead of `slow_kwargs`

As commented in https://github.com/rapidsai/cudf/pull/16266#issuecomment-2234003218 I am not sure this is _exactly_ a bug